### PR TITLE
Update product pages

### PR DESCRIPTION
**Pull request by stevenjmesser**
After chatting with @psd last week, I've raised this PR to meet some needs of our product pages:

1. speak about the data design and data provision capabilities on the homepage, so that external parties understand how we help make the data a reality
2. add the Organisations page back to the header navigation, so that external parties can see the range of organisations we work with
3. rename the navigation item for data providers, so that it fits the style of other nav items
4. reorder the guidance for data providers, to encourage them to use the Check service sooner
5. add content explaining the improvements we're making to the Check service (thank you to @CharliePatterson for his help)

I need some help checking it all works and doesn't break any tests. Instead of downloading all dependencies and building the databases so that I could edit some static pages, I skipped some steps in the README. The only commands I've used are `make init`, `make frontend` and `make server`, so I'm wondering whether I might have broken some tests? Please check and let me know!

### CODE REVIEW

File: application/templates/pages/guidance/index.html [hunk 1]
```diff
@@ -12,7 +12,43 @@
 {% block content_primary %}
 
     <p>
-   Guidance for planning authorities adding data to the platform.
+      We’re working with 73 local planning authorities (LPAs) through the <a class="govuk-link" href="https://opendigitalplanning.org/community-members" target="_blank">Open Digital Planning community</a> to help them provide data to our platform. There are four steps to follow:
```
Comment (gwencariad) [inline application/templates/pages/guidance/index.html:15]: change four to 4

File: application/templates/pages/guidance/index.html [hunk 2]
```diff
@@ -12,7 +12,43 @@
...
+    </p>
+
+    <ul class="govuk-list govuk-list--bullet">
+      <li>prepare data to the specifications</li>
```
Comment (gwencariad) [inline application/templates/pages/guidance/index.html:19]: number these out as a list, 1 to 4

File: application/templates/pages/guidance/try-check-publish-service.md [hunk 3]
```diff
@@ -1,4 +1,6 @@
-We have an online service that checks if some of your data meets the standard and helps you publish it. 
+We have an online service that checks if your data meets the technical specification and helps you publish it.
+
+You can use the service to check your existing datasets against the specifications, which will help you plan your project. You can also use it after preparing the data, before you publish the data on your website.
```
Comment (gwencariad) [inline application/templates/pages/guidance/try-check-publish-service.md:3]: take out the second 'the data' so it reads:

You can also use it after preparing the data, before you publish it on your website.

File: application/templates/pages/guidance/index.html (hunk 2 above)
Comment (stevenjmesser) [reply application/templates/pages/guidance/index.html:19]: Made some other changes to follow the guidance for steps: https://www.gov.uk/guidance/style-guide/a-to-z-of-gov-uk-style#bullet-points-steps

**Review by cjohns-scottlogic — APPROVED** [review]

**Review by cjohns-scottlogic — APPROVED** [review]

**Pull request merged**